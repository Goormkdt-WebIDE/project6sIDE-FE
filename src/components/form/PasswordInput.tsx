import React from "react";
import {
  FieldValues,
  Path,
  UseFormRegister,
  UseFormReturn,
} from "react-hook-form";

type Props<T extends FieldValues> = {
  register: UseFormRegister<T>;
  errors: UseFormReturn<T>["formState"]["errors"];
  placeholder?: string;
  name: keyof T;
};

function PasswordInput<T extends FieldValues>({
  register,
  errors,
  placeholder,
  name,
}: Props<T>) {
  return (
    <>
      <input
        type="password"
        placeholder={placeholder ? placeholder : "Password"}
        {...register(name as Path<T>, {
          required: true,
          minLength: 6,
        })}
        className="border-none rounded-md p-2 w-full mt-4"
      />
      {errors[name] && errors[name]?.type === "required" && (
        <p className="text-red-500" role="alert">
          <span className="inline-block align-middle">âš  </span>
          This password field is required
        </p>
      )}
      {errors[name] && errors[name]?.type === "minLength" && (
        <p className="text-red-500" role="alert">
          Password must have at least 6 characters
        </p>
      )}
    </>
  );
}

export default PasswordInput;
