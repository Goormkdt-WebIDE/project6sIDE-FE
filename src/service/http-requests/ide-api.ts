import axios from "axios";
import { BASE_URL } from "../../common/constant";

export type FormValue = {
  name: string;
  email: string;
};

const route = {
  CREATE_NEW_PROJECT: BASE_URL + "/api/projects/save",
  GET_PROJECT: BASE_URL + "/api/projects",
  ADD_ROOT_DIRECTORY(projectId: string) {
    return BASE_URL + `/api/projects/v2/${projectId}/directories`;
  },
  ADD_ROOT_CODE(projectId: string) {
    return BASE_URL + `/api/projects/v2/${projectId}/code`;
  },
  DELETE_DIRECTORY(directoryId: string) {
    return BASE_URL + `/api/projects/v2/directories/${directoryId}`;
  },
  DELETE_CODE(codeId: string) {
    return BASE_URL + `/api/projects/v2/${codeId}/code`;
  },
  DELETE_PROJECT(projectId: string) {
    return BASE_URL + `/api/projects/${projectId}`;
  },
};

export async function createNewProject(data: FormValue) {
  return axios.post(route.CREATE_NEW_PROJECT, {
    name: data.name,
    email: data.email,
  });
}

export async function getProject(data: FormValue) {
  return axios.post(route.GET_PROJECT, {
    email: data.email,
    projectName: data.name,
  });
}

export async function addRootDirectory(name: string, projectId: string) {
  return axios.post(route.ADD_ROOT_DIRECTORY(projectId), {
    name,
  });
}

export async function addRootCode(
  name: string,
  text: string,
  projectId: string
) {
  return axios.post(route.ADD_ROOT_CODE(projectId), {
    name,
    text,
  });
}

export async function deleteDirectory(directoryId: string) {
  return axios.delete(route.DELETE_DIRECTORY(directoryId));
}

export async function deleteCode(codeId: string) {
  return axios.delete(route.DELETE_CODE(codeId));
}

export async function deleteProejct(projectId: string) {
  return axios.delete(route.DELETE_PROJECT(projectId));
}
