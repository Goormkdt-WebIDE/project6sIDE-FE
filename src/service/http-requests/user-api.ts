import axios from "axios";
import { BASE_URL } from "../../common/constant";

export type FormValue = {
  name: string;
  email: string;
  password: string;
  password_confirm: string;
};

const route = {
  LOGIN: BASE_URL + "/user/login",
  REGISTER: BASE_URL + "/user/signUp",
  PASSWORD_RESET: BASE_URL + "/user/resetPassword",
};

export async function login(data: FormValue) {
  return axios.post(route.LOGIN, {
    email: data.email,
    password: data.password,
  });
}

export async function register(data: FormValue) {
  return axios.post(route.REGISTER, {
    username: data.name,
    email: data.email,
    password: data.password,
  });
}

export async function passwordReset(data: FormValue) {
  return axios.post(route.PASSWORD_RESET, {
    email: data.email,
    password: data.password,
    newPassword: data.password_confirm,
  });
}
