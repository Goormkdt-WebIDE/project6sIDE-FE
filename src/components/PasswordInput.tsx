import React from "react";
import { UseFormRegister, UseFormReturn } from "react-hook-form";
import { FormValue } from "../service/http-requests/user-api";

type Props = {
  register: UseFormRegister<FormValue>;
  errors: UseFormReturn<FormValue>["formState"]["errors"];
  placeholder?: string;
  register_type?: "password" | "password_confirm";
};

const PasswordInput = ({
  register,
  errors,
  placeholder,
  register_type,
}: Props) => {
  return (
    <>
      <input
        type="password"
        placeholder={placeholder ? placeholder : "Password"}
        {...register(register_type ? register_type : "password", {
          required: true,
          minLength: 6,
        })}
        className="border-none rounded-md p-2 w-full mt-4"
      />
      {errors.password && errors.password.type === "required" && (
        <p className="text-red-500">
          <span className="inline-block align-middle">âš  </span>
          This password field is required
        </p>
      )}
      {errors.password && errors.password.type === "minLength" && (
        <p className="text-red-500">Password must have at least 6 characters</p>
      )}
    </>
  );
};

export default PasswordInput;
