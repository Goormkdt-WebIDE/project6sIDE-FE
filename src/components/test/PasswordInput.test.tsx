import "@testing-library/jest-dom";
import PasswordInput from "../form/PasswordInput";
import { FieldError } from "react-hook-form";
import { render, screen } from "@testing-library/react";
import React from "react";
import { FormValue } from "../../service/http-requests/user-api";

describe("PasswordInput", () => {
  const register = jest.fn();

  afterEach(() => {
    register.mockReset();
  });

  it("스냅샷 테스트", () => {
    const { asFragment } = renderPasswordInput();
    expect(asFragment()).toMatchSnapshot();
  });

  it("placeholder prop을 전달하면 그것을 올바르게 render 해야한다", () => {
    const placeholder = "Password를 입력하세요";
    renderPasswordInput(placeholder);
    expect(screen.getByPlaceholderText(placeholder)).toBeInTheDocument();
  });

  it("required 에러가 발생하면 밑에 에러가 표시되어야한다", () => {
    renderPasswordInput("", { type: "required" } as FieldError);
    expect(screen.getByRole("alert")).toBeInTheDocument();
  });

  it("minLength 에러가 발생하면 밑에 에러가 표시되어야한다", () => {
    renderPasswordInput("", { type: "minLength" } as FieldError);
    expect(screen.getByRole("alert")).toBeInTheDocument();
  });

  function renderPasswordInput(placeholder?: string, error?: FieldError) {
    return render(
      <PasswordInput<FormValue>
        register={register}
        errors={{ password: error }}
        placeholder={placeholder}
        name="password"
      />
    );
  }
});
