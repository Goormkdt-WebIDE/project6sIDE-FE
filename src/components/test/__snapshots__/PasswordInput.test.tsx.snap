// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput 스냅샷 테스트 1`] = `
<DocumentFragment>
  <input
    class="border-none rounded-md p-2 w-full mt-4"
    placeholder="Password"
    type="password"
  />
</DocumentFragment>
`;
